### PR TITLE
VAN-2340 App integration pipeline modules for code scanning

### PR DESCRIPTION
**Pull request by pranaySinghDev**
Add CI pipeline modules to scan vulnarable go code
using https://github.com/securego/gosec and
https://github.com/golangci/golangci-lint

### CODE REVIEW

**Review by dshelley66 — COMMENTED** [review]
Can you add same module for AWS?

File: pipeline-modules/app-service/gcp/go-scan.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      type: array
+      items:
+        type: string
+    working_dir:
+      title: Working directory
+      description: The current working directory to execute command
+      type: string
+      default: repo
+  required:
+    - go_sec_args
```
Comment (dshelley66) [inline pipeline-modules/app-service/gcp/go-scan.yaml:34]: why are these required? Isn't it possible to run these tools without any args?

File: pipeline-modules/app-service/azure/go-scan.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  required:
+    - go_sec_args
+    - go_lint_args
+  internal:
+    - working_dir
+template: |
+  steps:
+  - script: |
+      echo Installing Gosec
+      wget -O - -q https://raw.githubusercontent.com/securego/gosec/master/install.sh | sh -s v2.9.6
```
Comment (dshelley66) [inline pipeline-modules/app-service/azure/go-scan.yaml:42]: should we parameterize this version #?

Comment (pranaySinghDev) [reply pipeline-modules/app-service/azure/go-scan.yaml]: although I have experienced some issues while working with older versions, I updated the template to have current versions defaulted  to variables.

File: pipeline-modules/app-service/gcp/go-scan.yaml (hunk 1 above)
Comment (pranaySinghDev) [reply pipeline-modules/app-service/gcp/go-scan.yaml]: Updated ! removed required and added default commands to make it work without template inputs. 
But to keep in mind the default behavior of these commands is to break the pipeline if there is an issue discovered by the template. extra args are needed to let the build pass without failing eg. `-no-fail`

File: pipeline-modules/app-service/gcp/go-scan.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      title: Working directory
+      description: The current working directory to execute command
+      type: string
+      default: repo
+  internal:
+    - working_dir
+template: |
+  steps:
+  - id: go_security
+    name: "securego/gosec"
```
[the remark this answers is not in the document]
Comment (pranaySinghDev) [reply pipeline-modules/app-service/gcp/go-scan.yaml]: thank you, it worked!

**Review by dshelley66 — APPROVED** [review]
